Source/WebKit/NetworkProcess/NetworkDataTaskBlob.cpp

```diff
@@ -142,6 +142,32 @@ void NetworkDataTaskBlob::resume()
     });
 }
 
+auto NetworkDataTaskBlob::adjustAndValidateRangeBounds() -> std::optional<Error>
```

Comment:
Could we refactor so we share one copy of this code with `BlobResourceHandle`?

Reply:
In this latest iteration, I introduced a new `BlobResourceHandleBase` base class to avoid adding new code duplication in this PR at least. In a follow-up, I will move more of the existing logic from the subclasses to `BlobResourceHandleBase`.